### PR TITLE
Add thread editing

### LINKED ISSUE #1510: Archiving and Locking Threads
Currently, it seems that it has not been implemented to allow bots to archive and lock threads. I have looked through the documentation and was unable to find any method to do so. I also asked in the discord server and it doesn't seem like anyone else knows of it so I thought that I create an issue because this is probably something that should be implemented. I presume it should be a pretty simple thing to implement.

### PR DESCRIPTION
**Pull request by lhjt**
This PR adds the `edit_thread` method to `Client`, `ChannelId`, and `GuildChannel`.

This allows the following properties of threads to be edited by this library:

* name
* auto-archive duration
* archived state
* locked state

This PR would also close #1510.

### CODE REVIEW

File: src/http/client.rs
Comment (lhjt) [inline src/http/client.rs]: I've just realised that this declaration is in the incorrect location in the file; I'll fix it.

**Comment by lhjt** [comment]
The CI check failed due to clippy being unable to properly run on forks (if my understanding is correct), though the warnings emitted in stdout indicate there are no issues due to my modifications to files.

Is there anything I still need to do for this to be merged?

**Comment by Milo123459** [comment]
> The CI check failed due to clippy being unable to properly run on forks (if my understanding is correct), though the warnings emitted in stdout indicate there are no issues due to my modifications to files.
> 
> Is there anything I still need to do for this to be merged?

#1518 this PR should fix the issues. Possibly wait until that is merged?

**Review by arqunis — APPROVED** [review]
Looks good. And, although `clippy` is being an inconvenience, it doesn't prevent this pull request from being merged.

**Pull request merged**
